### PR TITLE
Add wide string search to rafind2

### LINKED ISSUE #1945: Add rafind2 -S to search for wide strings

### PR DESCRIPTION
**Pull request by NickStephens**
closes #1945

### CODE REVIEW

File: libr/search/keyword.c [hunk 1]
```diff
@@ -53,6 +53,46 @@ R_API RSearchKeyword* r_search_keyword_new_str(const char *kwbuf, const char *bm
...
+		bmlen = r_hex_str2bin (bmstr, bmbuf);
+		if (bmlen < 1) {
+			free(bmbuf);
+			bmbuf = NULL;
+		}
+	}
+
+	len = strlen(kwbuf);
+	str = malloc((len+1)*2);
+	for (p2=kwbuf, p=str; *p2; p+=2, p2++)
```
Comment (crowell) [inline libr/search/keyword.c]: `{` brackets go on same line as for

File: libr/search/keyword.c [hunk 2]
```diff
@@ -53,6 +53,46 @@ R_API RSearchKeyword* r_search_keyword_new_str(const char *kwbuf, const char *bm
 	return kw;
 }
 
+R_API RSearchKeyword* r_search_keyword_new_wide(const char *kwbuf, const char *bmstr, const char *data, int ignore_case)
+{
```
Comment (crowell) [inline libr/search/keyword.c]: `{` go on same line as fn decl

File: libr/search/keyword.c [hunk 3]
```diff
@@ -53,6 +53,46 @@ R_API RSearchKeyword* r_search_keyword_new_str(const char *kwbuf, const char *bm
...
+	for (p2=kwbuf, p=str; *p2; p+=2, p2++)
+	{
+		if (ignore_case)
+			p[0] = tolower((const unsigned char)*p2);
+		else
+			p[0] = *p2;
+		p[1] = 0;
+	}
+
+	kw = r_search_keyword_new ((ut8 *)str, len*2, bmbuf,
```
Comment (crowell) [inline libr/search/keyword.c]: can this all fit on one line?

**Comment by crowell** [comment]
lgtm modulo a few nits :+1:

**Comment by radare** [comment]
Looks good! thanks

**Pull request closed without merging**